### DISCUSSION TITLE
zh/tasks/classify/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#9716] [Docs]
# zh/tasks/classify/

了解用于图像分类的YOLOv8 Classify 模型。获取有关预训练模型列表以及如何训练、验证、预测和导出模型的详细信息。

https://docs.ultralytics.com/zh/tasks/classify/

**Comment by abcc235** [comment]
The above video describes a situation where there is only one class in an image. If an image has multiple class objects, which class folder should the image be placed in?

**Reply by pderrenger** [reply]
@abcc235 in image classification tasks, each image should be placed in the folder corresponding to the class that represents the primary subject or the most significant object of the image. If an image contains multiple objects from different classes, you should consider the context and the main focus of the image to decide its class. If the image equally represents multiple classes, you might need to duplicate the image into multiple class folders or choose a more general class that encompasses all objects if applicable. This approach helps maintain clarity and consistency in your dataset.

**Comment by hoobnn** [comment]
我是否可以训练一个YOLO分类模型，来完成对图像质量的分类，例如清晰度：清晰或者不清晰，曝光度：过高、过低或者正常？

**Reply by glenn-jocher** [reply]
@TrioTea yes, you can train a YOLO classification model for image quality assessment by organizing your dataset with folders like `clear/` and `unclear/` for clarity, and `overexposed/`, `underexposed/`, `normal/` for exposure. Use the `yolo classify train` mode with a `yolo11-cls` model, following our classification training guide at https://docs.ultralytics.com/tasks/classify.

**Comment by hoobnn** [comment]
Can I train a YOLO classification model to complete the classification of image quality, such as clarity: clear or unclear, exposure: too high, too low or normal?

**Reply by glenn-jocher** [reply]
@TrioTea yes, you can train a YOLO classification model for image quality assessment by preparing a custom dataset with classes like "clear/unclear" and "normal/high/low exposure". Structure your dataset directories with these class labels as subfolders, then train using `yolo classify train` as shown in our classification documentation. For commercial use, ensure compliance with our licensing terms at https://ultralytics.com/license.

**Comment by question7603** [comment]
Can I use YOLO cls model to try extracting embedding of an image, like a 512 or 1024 dim array so that I can finish the retrieval task?

**Reply by glenn-jocher** [reply]
@question7603 yes, you can extract image embeddings using YOLO classification models. After running predictions with `model.predict()`, access the `probs` attribute of the results object to obtain feature vectors. For example:

```python
from ultralytics import YOLO
model = YOLO('yolo11n-cls.pt')
results = model('path/to/image.jpg')
embeddings = results[0].probs  # Shape: (num_classes,)
```

This provides the final layer embeddings before classification. For intermediate layer features, you may need to modify the model architecture slightly. Let us know if you need further guidance!

**Comment by hellotenet** [comment]
According to my observation, during the training of YOLO for classification tasks, it randomly selects a portion of data from the test set to evaluate the accuracy of this round. If I want to use all the data in the test set for testing in each epoch and provide top_acc1 and top_acc5, which parameter should I add in the yaml file?

**Comment by wowoso** [comment]
用图像分类任务无法获得F1分数、map、recall、precession等指标怎么办

**Comment by wowoso** [comment]
What if we can't obtain metrics such as F1 score, mAP, recall, and precision in image classification tasks?

**Comment by jiangshouye** [comment]
如何指定分类标签的次序呢？训练完成后，如何获得模型里的标签顺序呢？

**Reply by glenn-jocher** [reply]
分类任务里类别索引顺序是从你的数据集目录结构推断出来的（通常按 `train/` 下各类别子文件夹名称排序生成），如果你想“指定顺序”，最稳妥的方法是把类别文件夹命名成可排序的形式（例如 `00_cat/`、`01_dog/`）；训练完成后直接读取模型里的 `model.names` 就能看到标签到索引的对应关系，参考 [图像分类文档](https://docs.ultralytics.com/zh/tasks/classify/)。

```python
from ultralytics import YOLO

model = YOLO("runs/classify/train/weights/best.pt")
print(model.names)  # {0: '00_cat', 1: '01_dog', ...}
```

**Comment by lixjohn** [comment]
If I can train Yolo-cls mode to identify the level of object moisture content? Such as coal? Thanks.

**Reply by glenn-jocher** [reply]
Yes, Ultralytics YOLO26 classification can work for coal moisture levels if you label them as discrete classes like `dry`, `medium`, and `wet` and train with the standard [classification dataset format](https://docs.ultralytics.com/datasets/classify/); if you need an exact moisture percentage rather than bins, that’s a regression problem, and if the visual differences are subtle you may need NIR/thermal imaging instead of RGB alone.

**Reply by lixjohn** [reply]
Thanks, I think classification is enough because we just need high level of coal moisture not exact percentage.

**Reply by glenn-jocher** [reply]
Yes, that should be a good fit for `classify`; use a few clear bins like `low`, `medium`, and `high`, keep lighting/camera distance as consistent as possible, and start from a pretrained `yolo26n-cls.pt` model—if you’d like, share a sample class layout or training command here and we can sanity-check it.

**Reply by lixjohn** [reply]
Thank you very much, that makes sense. I just start project and not go to training phase yet, but I am happy to share sample and training command once I have to get more help on this.

**Reply by glenn-jocher** [reply]
Sounds good—when you reach training, share your dataset folder structure and exact `yolo classify train ...` command here, and we’ll help sanity-check it against the [classification guide](https://docs.ultralytics.com/tasks/classify/).

**Reply by lixjohn** [reply]
Thank you! 
I got another question related to segment, hopefully I can get your suggestion as well, it is a little bit tricky :-). The coal is very small ( around 10mm one piece), but we have identify them in a truck (10m*3m) in order to ensure the truck carry right size of coal, one picture for one truck, it means there are huge member of small coal in one picture, I don't have image resolution yet, but I suppose it is clear enough to show details for each coal, my questions are
1. how to label data before training, I can use SAM to label each coal but I have to say it is impossible to label all coal in picture because the coal member is huge, if I can just label some of them in pictures? I think we can use around 300 pictures to start training, but not all coal labeled in one picture. Does it will impact accuracy of model?
2. Is it possible for Yolo26-segment to identify each coal in this situation from algorithm and performance point of view? In one picture, the truck is big but carry large member of similar small objects (10mm vs. 10m). Any solution we can take for this situation?

Apricate for any suggestions...

**Reply by glenn-jocher** [reply]
Yes, but you should not leave visible coals unlabeled in training images because the model will treat them as background and recall will drop; for this kind of dense small-object problem, the practical approach is to split each truck image into smaller overlapping tiles, fully annotate every visible coal inside the selected tiles, and train on those crops rather than the full frame. If each coal is still only a few pixels after tiling, `segment` will struggle and `detect` or even patch-level `classify` may be a better fit, and our [data collection and annotation guide](https://docs.ultralytics.com/guides/data-collection-and-annotation) is a good reference—feel free to post one sample image here when you have it and we can suggest the best labeling strategy.
